pinot-broker/src/main/java/org/apache/pinot/broker/broker/helix/HelixBrokerStarter.java

```diff
@@ -330,6 +334,27 @@ public void start()
     LOGGER.info("Finish starting Pinot broker");
   }
 
+  @VisibleForTesting
+  static void updateHelixHost(PinotConfiguration brokerConf, HelixManager helixManager, String clusterName, String brokerInstanceId) {
+    boolean allowUpdateHelixHost = brokerConf.getProperty(Broker.BROKER_DYNAMIC_HELIX_HOST, false);
```

Comment:
I don't think we need this config. If host/port does not match instance id, we should always update them

Reply:
Removed.